src/object_store/src/object/mod.rs

```diff
@@ -792,7 +792,10 @@ pub async fn build_remote_object_store(
 ) -> ObjectStoreImpl {
     match url {
         s3 if s3.starts_with("s3://") => {
-            if std::env::var("RW_USE_OPENDAL_FOR_S3").is_ok() {
+            if std::env::var("RW_USE_OPENDAL_FOR_S3")
+                .map(|v| v != "false")
+                .unwrap_or(true)
```

Comment:
let's have some util function for bool env check ([example](https://github.com/risingwavelabs/risingwave/blob/7722cdb040028331b16043a4e5e15452d0235f50/src/common/src/util/env_var.rs#L21)). To avoid surprises, we should use case insensitive comparison.